nixio.spec.in

```diff
@@ -0,0 +1,61 @@
+Name:		nixio
+Version:	${VERSION_FULL}
+Release:	0.${RPM_RELTAG}%{?dist}
+Summary:	IO-libray for nix data files
+
+License:	BSD
+URL:		https://www.g-node.org/nix
+Source0:	%{name}-%{version}.tar.gz
+
+BuildRequires:	cmake, boost-devel, hdf5-devel, cppunit-devel
+
+%description
+Neuroscience information exchange - data model for annotated (neuroscience)
+data. For more information visit https://github.com/G-Node/nix
+
+%package devel
+Summary: nixio development files
+Group: Development/Libraries
+Requires: %{name} = %{version}-%{release}
+Requires: hdf5-devel, boost-devel
+
+%description devel
+nixio development headers and libraries.
+
+%prep
+%setup -q -n %{name}-%{version}
+
+%build
+mkdir -p build
+cd build
+%cmake .. \
+       -DCMAKE_INSTALL_PREFIX=/usr \
+       -DCMAKE_BUILD_TYPE=RelWithDebInfo
+make %{?_smp_mflags}
+
+%install
+cd build
+make install DESTDIR=%{buildroot}
+
+%post -p /sbin/ldconfig
+
+%postun -p /sbin/ldconfig
+
+
+%files
+%license LICENSE LICENSE.h5py
+%doc README.md CONTRIBUTING.md
+%{_bindir}/nixio-tool
+%{_libdir}/libnixio.so.1*
+
+%files devel
+%{_includedir}/nixio-1.0/nix.hpp
```

Comment:
stupid question demonstrating complete ignorance: Where does the 1.0 come from and if it needs to be incremented, shouldn't we make a variable out of it?

Reply:
I guess it is some form of new versioning (again, to be even more confusing). It is parallel for what we do with the header installation already (installing it into `nixio-1.0`) The idea would be to change that every time we want to break ABI and have it installable in parallel with other versions of NIX. As I hope we wan't break ABI between 1.5 and 1.6 using nixio-1.5 would look funny.
We could indeed define a new variable for both of these places.